pkg/monitor/agent/monitor.go

```diff
@@ -95,54 +99,90 @@ func NewMonitor(ctx context.Context, nPages int, server1_2 net.Listener) (m *Mon
 		},
 	}
 
-	// start new MonitorListener handler
-	go m.connectionHandler1_2(ctx, server1_2)
+	return a, nil
+}
+
+// SendEvent distributes an event to all monitor listeners
+func (a *Agent) SendEvent(typ int, event interface{}) error {
+	var buf bytes.Buffer
+
+	if a == nil {
+		return fmt.Errorf("monitor agent is not set up")
+	}
+
+	if err := gob.NewEncoder(&buf).Encode(event); err != nil {
+		return fmt.Errorf("Unable to gob encode: %w", err)
+	}
+
+	p := payload.Payload{Data: append([]byte{byte(typ)}, buf.Bytes()...), CPU: 0, Lost: 0, Type: payload.EventSample}
```

Comment:
instead of doing this append, can't we initialize the buffer with the `typ`?

`buf.WriteByte(byte(typ))`

Reply:
Sure, will do. FWIW, this code is copy-pasted from the old implementation, I did not perform any cleanups.